src/osemgrep/cli_interactive/Interactive_subcommand.ml

```diff
@@ -36,12 +36,32 @@ type command =
   (* actual commands *)
   | Exit
 
-(* interactive formula *)
-type iformula =
+(* interactive formula, but in a zipper!
+
+   When in Pattern Builder mode, we want to be able to traverse the
+   formula, up and down. This means we need another zipper. We can
+   use `Framed_zipper` of frame 1 to mimic a true zipper, again.
```

Comment:
Maybe define a Zipper.ml where you simply reuse Framed_zipper.ml but set the frame to 1?